code/answering_machine_detection/README.md

```diff
@@ -0,0 +1,96 @@
+# Answering Machine Detection
+This guide utilizes Answering Machine Detection to determine whether a human or voicemail machine has picked up the phone so that it can either dial a number to connect someone to the human or leave a message for the voicemail box. 
+
+When a call is initiated with ```machine_detection``` set to ```DetectMessageEnd```, there is a parameter posted to the webhook called AnsweredBy. The potential options here are machine_end_other, machine_end_beep, machine_end_silence, and human. If the parameter returned is anything other than human, this example will take a short pause to wait for the beep and then play a message for the intended recipient. If the parameter returned is human, it will dial a number to connect the human to another person.
+
+We start with the necessary imports below and set the port to 8080. 
+
+```ruby
+require 'rubygems'
+require 'sinatra'
+require 'signalwire'
+
+set :port, 8080
+```
+
+Here we create a route for Sinatra and initialize response as a VoiceResponse object so that we can use Say, Dial, Hangup, and Pause. 
+
+
+```ruby
+post '/start' do
+  response = Signalwire::Sdk::VoiceResponse.new
+```
+
+Now we use a switch statement to designate that we will take different actions depending on the value of ```[:AnsweredBy]```. You can see below that if anything other than human is returned, this code will output "It's a machine", take a brief pause to allow the voicemail system time to reach the recording beep, leave a voicemail for the intended recipient, and hang up. If the value returned is human, this code will output "We got ourselves a live human here!" and then dial the number of your agent (or the next part of your standard call flow). In the dial, you have the option to either hard code the number you would like to dial or use environment variables. 
+
+
+```ruby
+case params[:AnsweredBy]
+  when 'machine_end_other', 'machine_end_silence', 'machine_end_beep'
+    puts "It's a machine!"
+    # put in a little pause to allow the recording to start on the other end
+    response.pause(length: 1)
+    # replace messsage with whatever voicemail you want to leave
+    response.say(message: "Hello! This is the County Medical Center. We are calling you to confirm your doctor appointment. Please call us back as soon as possible.")
+    response.hangup
+  when 'human'
+    puts "We got ourselves a live human here!"
+    response.dial do |dial|
+      # defaulting to calling a time voice announcement number as an example
+      dial.number(ENV.fetch('AGENT_NUMBER', '+12027621401'))
+    end
+  end
+```
+
+Lastly, we need to put in code for the sake of debugging the LaML. LaML requires that proper XML is returned or it will not work correctly. 
+```ruby
+  puts response.to_s
+  response.to_s
+```
+
```

Comment:
Add the corresponding LAML generated by the human and machine so people that are not on Ruby can tell what is going on.